File: client/src/app/core/api/requests/register.request.ts
import { UserTier } from '../../enums/user-tier.enum';
import { Gender } from '../../enums/gender.enum';
import { Region } from '../../enums/region.enum';

export interface RegisterRequest {
  firstName: string;
  lastName: string;
  username: string;
  email: string;
  password: string;
  tier: UserTier;
  gender: Gender;
  region: Region;
}

File: backend/Application/Contracts/Requests/RegisterRequest.cs
using CoachApi.Domain.Enums;

namespace CoachApi.Application.Contracts.Requests;

public class RegisterRequest
{
    public required string FirstName { get; set; }
    public required string LastName { get; set; }
    public required string Username { get; set; }
    public required string Email { get; set; }
    public required string Password { get; set; }
    public required UserTier Tier { get; set; }
    public required Gender Gender { get; set; }
    public required Region Region { get; set; }
}

